### ISSUE TITLE
[Feature Request]: Show ComboBox clear icon only on hover/ focus of input element

### ISSUE THREAD

**Issue by senthil-kumaran-ibm** [carbon-design-system/carbon#20317]
### The problem

Consider I have 3 ComboBoxes in a form, and every field with a cross icon makes it look cluttered. So, it would be great if it were revealed only on hover or focus of the input field.

### The solution

Simple CSS correction to show and hide the Clear icon on hover or focus

### Examples

https://github.com/user-attachments/assets/3c973274-0569-4191-896c-14a27bf4daa4

### Design impact

Yes, both visual and interaction

### Documentation impact

- [ ] The website documentation (carbondesignsystem.com)
- [x] Figma kit documentation
- [x] Storybook(s) documentation
- [ ] I'm not sure

### Package

@carbon/react

### Application/PAL

API Connect/Studio

### Business priority

Medium Priority = upcoming release but is not pressing

### Available extra resources

I can contribute this change.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)

**Comment by adamalston** [comment]
Would it be an accessibility issue to make a control visible only on hover?

**Comment by tay1orjones** [comment]
I agree, this is probably going to fail a success criterion, I'm just not sure which one. We looked at [focus not obscured](https://www.w3.org/WAI/WCAG22/Understanding/focus-not-obscured-minimum.html),  [meaningful sequence](https://www.w3.org/WAI/WCAG21/Understanding/meaningful-sequence.html), and [focus visible](https://www.w3.org/WAI/WCAG22/Understanding/focus-visible.html) but none mention this concern specifically. In any case I think there's also UX concerns with this negatively impacting discoverability. Both for keyboard users but also mouse users.

@tombrunet could you help pinpoint what SC this might impact or relate to?

**Comment by tombrunet** [comment]
The only SC that I can think of offhand that would be related is https://www.w3.org/TR/WCAG22/#content-on-hover-or-focus. It should have its own screen real-estate (not be a layer over the input), should persist on hover of both the input and the additional button, so I'm pretty sure you would meet that.

That said, keep in mind that you can't really "hover" with a mobile device. I think you can get away with it provided that you show the clear on both hover AND focus (note that the video is hiding the clear button even though the input still has focus).

**Comment by tay1orjones** [comment]
Based on that, specifically the limitations of mobile, we're not going to move forward with this idea. If you'd like to still incorporate this into your app you should be able to do so with custom styling and events.
